src/TheThingsNetwork.cpp

```diff
@@ -401,13 +401,12 @@ void TheThingsNetwork::autoBaud()
 
 void TheThingsNetwork::reset(bool adr)
 {
-  if (!baudDetermined)
-  {
-    autoBaud();
-  }
-
+  autoBaud();
   size_t length = readResponse(SYS_TABLE, SYS_RESET, buffer, sizeof(buffer));
 
+  autoBaud();
+  length = readResponse(SYS_TABLE, SYS_TABLE, SYS_GET_VER, buffer, sizeof(buffer));  
```

Comment:
It is called in the autobaud function to make sure the autobaud was successful and that communication with the radio is good. If we remove it there we won't know if the autobaud was successful or not. Can you think of a better way to handle this?